### PR TITLE
SAAS-1038: New activation page

### PR DESCRIPTION
**Pull request by TomasBlazek3PG**
image

### CODE REVIEW

File: src/core/api/endpoints.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ export const ENDPOINTS = {
     SignOut: '/v1/auth/SignOut',
     SignUp: '/v1/auth/SignUp',
     UpdateProfile: '/v1/auth/UpdateProfile',
+    ActivateProfile: '/v1/auth/UpdateProfile',
```
Comment (fabio-silva) [inline src/core/api/endpoints.ts:10]: same as the one above?

File: src/pages/Activation/Activation.messages.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    description: 'Create your account in order to get access to Percona Portal.',
+    name: 'Name',
+    surname: 'Surname',
+    yourPassword: 'Your password',
+    verifyYourPassword: 'Verify your password',
+  },
+  errors: {
+    password: {
+      lowerCase: 'Password needs to contain a lowercase letter',
+      upperCase: 'Password needs to contain a uppercase letter',
```
Comment (fabio-silva) [inline src/pages/Activation/Activation.messages.ts:14]:
```suggestion
      upperCase: 'Password needs to contain an uppercase letter',
```

File: src/pages/Activation/Activation.tsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+import { getStyles } from './Activation.styles';
+import { Messages } from './Activation.messages';
+
+const Activation = () => {
+  const styles = useStyles(getStyles);
+  const [activateUser, { status, isLoading, error: activationError }] = useActivateProfileMutation();
+  const history = useHistory();
+  const { search } = window.location;
+  const formRef: React.MutableRefObject<FormApi | null> = useRef(null);
+  const query = useMemo(() => new URLSearchParams(search), [search]);
```
Comment (fabio-silva) [inline src/pages/Activation/Activation.tsx:25]: Do we really need `useMemo` for this? It seems an overkill, given that all those `q<name>` consts are derived from it and aren't changing when it changes as well.
We're using `search` as dependency. Is it gonna change unless we actually refresh the page, in this case? If so, `useMemo` brings no value.

File: src/pages/Activation/Activation.tsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+                          <span className={styles.infoIcon}>&#9432;</span>Your password should:
+                        </p>
+                        <ul className={styles.passwordList}>
+                          <li>Contain at least 10 characters</li>
+                          <li>Contain a lowercase letter</li>
+                          <li>Contain an uppercase letter</li>
+                          <li>Contain a number</li>
+                          {/* <li>Not contain parts of your username</li> */}
+                          <li>Not include your first or last name</li>
+                          <li>Not be the one of your last 4 passwords</li>
```
Comment (fabio-silva) [inline src/pages/Activation/Activation.tsx:159]:
```suggestion
                          <li>Not be one of your last 4 passwords</li>
```

**Review by fabio-silva — CHANGES_REQUESTED** [review]

**Review by fabio-silva — DISMISSED** [review]

**Review by MaksimMalakhovSimbirsoft — APPROVED** [review]

**Review by MaksimMalakhovSimbirsoft — APPROVED** [review]

**Pull request merged**